python/GafferSceneUI/SetEditor.py

```diff
@@ -0,0 +1,379 @@
+				columns = [
+					_GafferSceneUI._SetEditorSetNameColumn(),
+					self.__setMembersColumn,
+				],
+				selectionMode = GafferUI.PathListingWidget.SelectionMode.Rows,
+				displayMode = GafferUI.PathListingWidget.DisplayMode.Tree,
+			)
+
+			self.__pathListing.dragBeginSignal().connectFront( Gaffer.WeakMethod( self.__dragBegin ), scoped = False )
+			self.__pathListing.contextMenuSignal().connect( Gaffer.WeakMethod( self.__contextMenuSignal ), scoped = False )
+			self.__pathListing.keyPressSignal().connect( Gaffer.WeakMethod( self.__keyPressSignal ), scoped = False )
+
+		self._updateFromSet()
+
+	def __repr__( self ) :
+
+		return "GafferSceneUI.SetEditor( scriptNode )"
+
+	def __firstValidScenePlug( self, node ):
+
+		for plug in GafferScene.ScenePlug.RecursiveOutputRange( node ) :
+			if not plug.getName().startswith( "__" ):
+				return plug
+		return None
+
+	def _updateFromSet( self ) :
+
+		GafferUI.NodeSetEditor._updateFromSet( self )
+
+		# Decide what plug we're viewing.
+		self.__plug = None
+		self.__plugParentChangedConnection = None
+		self.__plugDirtiedConnection = None
+		node = self._lastAddedNode()
+		if node is not None :
+			self.__plug = self.__firstValidScenePlug( node )
+			if self.__plug is not None :
+				self.__plugParentChangedConnection = self.__plug.parentChangedSignal().connect(
+					Gaffer.WeakMethod( self.__plugParentChanged ), scoped = True
+				)
+				self.__plugDirtiedConnection = node.plugDirtiedSignal().connect(
+					Gaffer.WeakMethod( self.__plugDirtied ), scoped = True
+				)
+
+		self.__updatePathListingPath()
+
+	def _updateFromContext( self, modifiedItems ) :
+
+		if any( not i.startswith( "ui:" ) for i in modifiedItems ) or any( ContextAlgo.affectsSelectedPaths( i ) for i in modifiedItems ) :
+			self.__updatePathListingPath()
+
+	@GafferUI.LazyMethod( deferUntilPlaybackStops = True )
+	def __updatePathListingPath( self ) :
+
+		if self.__plug is not None :
+			# We take a static copy of our current context for use in the SetPath - this prevents the
+			# PathListing from updating automatically when the original context changes, and allows us to take
+			# control of updates ourselves in _updateFromContext(), using LazyMethod to defer the calls to this
+			# function until we are visible and playback has stopped.
+			contextCopy = Gaffer.Context( self.getContext() )
+			self.__searchFilterWidget.setScene( self.__plug )
+			self.__searchFilterWidget.setContext( contextCopy )
+			self.__pathListing.setPath( GafferScene.SetPath( self.__plug, contextCopy, "/", filter = self.__filter ) )
+		else :
+			self.__pathListing.setPath( Gaffer.DictPath( {}, "/" ) )
+
+	def _titleFormat( self ) :
+
+		return GafferUI.NodeSetEditor._titleFormat( self, _maxNodes = 1, _reverseNodes = True, _ellipsis = False )
+
+	def __plugDirtied( self, plug ) :
+
+		if self.__plug is not None and isinstance( plug, Gaffer.PathMatcherDataPlug ) and plug == self.__plug["set"] :
+			self.__updatePathListingPath()
```

Comment:
I don't think this should be necessary - see comment in `SetPath::plugDirtied()`.

Reply:
As above, removed in 4c99596.